### DISCUSSION TITLE
TEXTAREA vs. TEXTBOX for smaller fields

### DISCUSSION THREAD

**Discussion by ghost** [OP-TED/eForms-SDK#547] [Q&A] [answered]
We are generating our frontend forms using the SDK-Files, esp. fields.json and the forms-json Files.
In the forms definitions (e.g. 16.json), for a lot of smaller fields which never should accept multiline values, the "displayType" is defined as "TEXTAREA". This maked the forms not very easy to use for our customers,
Examples: BT-5121-Procedure, BT-5131-Procedure, BT-21-Procedure or practically all fields with maxLength <= 400.

In the eNotices2 tool, these fields are rendered as "INPUT"-elements, not as TEXTAREA. Why is this different to the forms definition? Is there any other information we can use for rendering the fields besides the forms-json Files?

**Comment by rouschr** [comment, marked as answer]
Hello/Hallo,

in the SDK export for the default displayType TEXTAREA the threshold is 30, so anything with max length > 30 is put as a TEXTAREA.
You can let your own application decide which one to use based on the "maxLength" of a field defined in the SDK fields.json.
Also consider using special inputs for various cases, for example if a field is of **type** "url", a horizontally longer input could make sense so that the end of the url is visible.
Note that the notion of TEXT and TEXTAREA is not fully tied to the way HTML works, as your application could also be a mobile app or something else.
